### PR TITLE
New getting started examples and faqs

### PR DESCRIPTION
**Pull request by minorgal**
We moved the original single example into a format very similar to python and Laura added all the new examples. She documented the examples and added FAQ's too.

### CODE REVIEW

File: javascript/gdx_getting_started_1.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  gdxDevice.close();
+  gdxDevice.sensors.filter();
+  -->  
+
+
+<!doctype html>
+<html>
+<head>
+  <title>Example 1: Showing Sensor Information</title>
+  <meta name="description" content="The gdx_getting_started_1.html example shows how you can use the gdx functions to list information on all of the sensor channels on a Go Direct device. This includes sensor channel number, sensor description, units, and 'incompatible sensors'">
```
Comment (stocktonkincade) [inline javascript/gdx_getting_started_1.html]: Looks like the "incompatible sensors" portion was removed from this example, so this comment is not accurate.

File: javascript/gdx_getting_started_1.html [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      } catch (err) {
+        console.error(err);
+          output.textContent += err;
+      }
+   
+    };
+    selectDeviceBtn.addEventListener('click', selectDevice);
+  </script>
+</body>
+</html>
```
Comment (stocktonkincade) [inline javascript/gdx_getting_started_1.html]: No newline

File: javascript/gdx_getting_started_2.html [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+  Press the 'Select GoDirect Device' button to open communication to the device in a Chrome browser. 
+  This example collects 10 data points and disconnects.
+  
+  This program uses the following functions from the GDX Library:
+  
+  gdxDevice.selectDevice();
+  gdxDevice.close();
+  gdxDevice.sensors.filter();
+  -->  
+
```
Comment (stocktonkincade) [inline javascript/gdx_getting_started_2.html]: Perhaps give this a similar title to the others. Like: For example 2: Simple Data Collection

File: javascript/gdx_getting_started_2.html [hunk 4]
```diff
@@ -1,3 +1,15 @@
+  <!--
+  This example uses the GDX JavaScript library to collect all of the sensor information from a Vernier GoDirect Sensor. 
+  Press the 'Select GoDirect Device' button to open communication to the device in a Chrome browser. 
+  This example collects 10 data points and disconnects.
+  
+  This program uses the following functions from the GDX Library:
+  
+  gdxDevice.selectDevice();
+  gdxDevice.close();
+  gdxDevice.sensors.filter();
```
Comment (stocktonkincade) [inline javascript/gdx_getting_started_2.html]: Also uses: enableDefaultSensors and demonstrates the use of "value-changed" events

**Comment by stocktonkincade** [comment]
Just some general nitpicking. ;)
- I see a number of uses of "GoDirect" in the comments and READMEs. We should be using "Go Direct", with the space between the two words
- The first occurrence of "Go Direct" on any README pages should be followed by the "®" symbol

File: javascript/gdx_getting_started_3.html [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+Press the 'Change Sensor and Collect Data' to change the sensor channel and collect 10 data points. 
+
+This program uses the following functions from the GDX Library:
+
+gdxDevice.selectDevice();
+gdxDevice.close();
+gdxDevice.sensors.filter();
+gdxDevice.getSensor(int);
+sensor.setEnabled(bool);
+-->
```
Comment (stocktonkincade) [inline javascript/gdx_getting_started_3.html]: Also uses: gdxDevice.start(); gdxDevice.stop()

File: javascript/gdx_getting_started_2.html [hunk 6]
```diff
@@ -1,3 +1,15 @@
+  <!--
+  This example uses the GDX JavaScript library to collect all of the sensor information from a Vernier GoDirect Sensor.
```
Comment (stocktonkincade) [inline javascript/gdx_getting_started_2.html]: For all of the examples other than getting_started_1, these top two lines of comments seem redundant.

File: javascript/gdx_getting_started_5.html [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  Press the 'Select GoDirect Device' button to open communication to the device in a Chrome browser.
+  Press the 'Download CSV' to save a copy of the data as a csv file.
+  
+  This program uses the following functions from the GDX Library:
+   
+  gdxDevice.selectDevice();
+  gdxDevice.close();
+  gdxDevice.sensors.filter();
+  gdxDevice.getSensor(int);
+  sensor.setEnabled(bool);
```
Comment (stocktonkincade) [inline javascript/gdx_getting_started_5.html]: Uses gdxDevice.start(). Does not use gdxDevice.sensors.filter()

File: javascript/gdx_getting_started_1.html (hunk 1 above)
Comment (minorgal) [reply javascript/gdx_getting_started_1.html]: I add it.

File: javascript/gdx_getting_started_5.html (hunk 7 above)
Comment (minorgal) [reply javascript/gdx_getting_started_5.html]: Done

File: javascript/readme.md [hunk 8]
```diff
@@ -2,20 +2,46 @@
...
-
-The go-direct-sensor-readout example is a simple example to get started with. It demonstrates how to get default enabled sensor(s) and log value changes.
-
-Click [here](./godirect-sensor-readout/) to run it in your browser, or [here](./godirect-sensor-readout/index.html) to view the source code.
+The examples in this folder will walk you through some of the basics of talking to a Go Direct device using JavaScript. 
+
+For other ideas and more advanced examples, take a look in the [/advanced_examples/](./advanced_examples) folder.
+
+## Example 1: Showing Sensor Information
+The [gdx_getting_started_1.html](./gdx_getting_started_1.html) example shows how you can use the gdx functions to list information on all of the sensor channels on a Go Direct device. This includes sensor channel number, sensor description, units, and "incompatible sensors".
```
Comment (stocktonkincade) [inline javascript/readme.md]: No incompatible sensors in this one. At least, not yet

Comment (minorgal) [reply javascript/readme.md]: Done

File: javascript/godirect-js-faqs.md [hunk 9]
```diff
@@ -8,5 +8,51 @@
...
+- [Visual Studio Code](https://code.visualstudio.com/Download)
+
+## How do I import the GDX library?
+ The easiest way to import a library is through a Content Delivery Network (CDN). This allows you to import libraries with a single line. To create programs with Go Direct compatibility put this in the header of your html file:
+
+ ``` html
+ <script src="https://unpkg.com/@vernier/godirect/dist/godirect.min.umd.js"></script> 
+ ```
+
+ ## How can I use the GDX library?
```
Comment (Smythics) [inline javascript/godirect-js-faqs.md]: space between lines

File: javascript/gdx_getting_started_1.html [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        const gdxDevice = await godirect.selectDevice();
+
+        // print name, serial number, order code and measurement period
+        output.textContent += `\n Connected to `+gdxDevice.name;
+        output.textContent += `\n Serial Number: `+gdxDevice.serialNumber;
+        output.textContent += `\n Order Code: `+gdxDevice.orderCode;
+        output.textContent += `\n measurementPeriod: `+gdxDevice.measurementPeriod ;
+      
+       
+        // wait for battery level and battery state and print
```
Comment (Smythics) [inline javascript/gdx_getting_started_1.html]: I get a timeout error here
 measurementPeriod: 500Error: write command timed out after 5s. Command: 10 Rolling Counter: f1
I think I saw code that corrected this from Jenny and Laura...but it is not here.

File: javascript/gdx_getting_started_3.html [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+  
+    const selectDeviceBtn = document.querySelector('#select_device');
+    const output = document.querySelector('#output');
+    let gdxDevice = "";
+   
+    const selectDevice = async () => {
+      try {
+        gdxDevice = await godirect.selectDevice();
+        output.textContent += `\n Connected to `+gdxDevice.name;
+        output.textContent += `\n Availiable sensors: (type the channel number into the alert box) `;
```
Comment (Smythics) [inline javascript/gdx_getting_started_3.html]: "Available" is spelled incorrectly. (extra "i")

File: javascript/gdx_getting_started_3.html [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+  <meta name="description" content="The gdx_getting_started_3.html example shows you how to use the library:
+Connect to a Go Direct device over BLE; Provide the user with a prompt to select the active sensors; Provide the user with a prompt to set the measurement period; Collect a few measurements from the enabled sensors; and
+Disconnect">
+  <meta charset="utf-8">
+  <script src="https://unpkg.com/@vernier/godirect/dist/godirect.min.umd.js"></script>
+</head>
+
+<body>
+  <button id="select_device">Select Go Direct Device</button>
+  <button id ="start_collection" onclick = "chooseSensor()">Change Sensor and Collect Data</button>
```
Comment (Smythics) [inline javascript/gdx_getting_started_3.html]: Suggest changing button label to read, "Select Sensor and Collect Data". 
Clarifies what user will do.

File: javascript/gdx_getting_started_3.html (hunk 11 above)
Comment (Smythics) [inline javascript/gdx_getting_started_3.html]: When I ran this I was looking for the alert box (wasn't visible yet). 

Suggest change to: (... Available sensors: (click on Select Sensor and Collect Data then type channel number into alert box) 

Or something less wordy, but equally clear.

File: javascript/gdx_getting_started_4.html [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+          sensor.setEnabled(true);
+       
+          // using the enabled sensor, collect 10 values
+          sensor.on('value-changed', (sensor) => {
+            // Only collect 10 samples and disconnect.
+            if (sensor.values.length > 10){
+              gdxDevice.close();
+            }
+            // log the sensor name, new value and units.
+            console.log(`Sensor: ${sensor.name} value: ${sensor.value} units: ${sensor.unit}`);
```
Comment (Smythics) [inline javascript/gdx_getting_started_4.html]: I can select a sensor, but I never see the data print out on the web page. Not sure what the issue is.

File: javascript/gdx_getting_started_5.html [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+    // create array to store data to be saved in the csv file
+    var data = [];
+
+    const selectDevice = async () => {     
+      try {
+        const gdxDevice = await godirect.selectDevice();     
+        
+        gdxDevice.start(1000);
+        
+        output.textContent += `\n Connected to `+gdxDevice.name;
```
Comment (Smythics) [inline javascript/gdx_getting_started_5.html]: When I run this I do not see the device name or any data. The .csv file is empty except for a header, "Sensor Values:"

File: javascript/godirect-js-faqs.md (hunk 9 above)
Comment (minorgal) [reply javascript/godirect-js-faqs.md]: thanks. fixed

File: javascript/gdx_getting_started_3.html (hunk 11 above)
Comment (minorgal) [reply javascript/gdx_getting_started_3.html]: Done

Comment (minorgal) [reply javascript/gdx_getting_started_3.html]: Done

File: javascript/gdx_getting_started_4.html (hunk 13 above)
Comment (minorgal) [reply javascript/gdx_getting_started_4.html]: sensor timeout on windows, likely a library issue.

File: javascript/gdx_getting_started_3.html (hunk 12 above)
Comment (minorgal) [reply javascript/gdx_getting_started_3.html]: done

File: javascript/gdx_getting_started_1.html (hunk 10 above)
Comment (minorgal) [reply javascript/gdx_getting_started_1.html]: sensor timeout on windows, likely a library issue.

**Pull request merged**